### PR TITLE
chore: Upgrade docz

### PR DESCRIPTION
**Pull request by seanhealy**
## Motivations

Old:

image

New:

image

### Added

- Atlantis is now better Jobber branded!
- Added meta tags to explain what Atantis is in social previews.

### Changed

- Upgraded Docz to v2.
  - This required a move to Gatsby and Gatsby themes.
    - We're now shadowing a few theme files to clean up the new Docs theme a bit as well as brand it a bit more. By default Docz uses [gatsby-theme-docz](https://github.com/doczjs/docz/tree/master/core/gatsby-theme-docz) as the theme.
    - New requirements around configuration means our web pack config is now nicely isolated into a Docs plugin.

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

Would be good to spot check pages on the Docs viewer. There weren't any changes to the components themselves as part of this.

## References

https://github.com/doczjs/docz/tree/v2.3.0-alpha.13/core/gatsby-theme-docz/src/components

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

**Comment by jcw-** [comment]
Isn't loading I don't think?
https://deploy-preview-205--jobber-atlantis.netlify.com/elevation

**Comment by jcw-** [comment]
https://atlantis.getjobber.com/components/select#option-props

Before:
image

After:
image

File: docs/getting-started-with-react/getting-started-with-react.mdx [hunk 1]
```diff
@@ -97,39 +97,30 @@ that.
 For this example we are going to build a card that allows you to enter a topic
 and then search for a related gif.
 
-import { Card } from "@jobber/components/Card";
-import { Content } from "@jobber/components/Content";
-import { Button } from "@jobber/components/Button";
-import { InputText } from "@jobber/components/InputText";
+import { Card } from "../../packages/components/src/Card";
```
Comment (seanhealy) [inline docs/getting-started-with-react/getting-started-with-react.mdx]: Can we use the proper aliases here now?

**Comment by eddysims** [comment]
I'm not sure how, but maybe we want to add in the description something about SSR. It has caught us a couple times with input time and now my PR that just went up in relation to the elevation page.

**Comment by seanhealy** [comment]
@eddysims I just noticed that the search icon is very tiny in Safari as well as the Edit / Github buttons are missing.

File: src/gatsby-theme-docz/components/Header/index.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import React from "react";
+
+export const Header = () => <></>;
```
Comment (JeffMarvin) [inline src/gatsby-theme-docz/components/Header/index.js]: I suspect this is just overriding the header to remove it. A brief comment to that effect will make this clear to future maintainers.

File: src/gatsby-theme-docz/components/NavLink/styles.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+export const link = theme => ({
+  ...styles.link,
+  ...theme.navigation.level2,
+});
+
+export const smallLink = theme => ({
+  ...styles.smallLink,
+  ...theme.navigation.level3,
+
+  "& + a:not([href*='#'])": {
```
Comment (JeffMarvin) [inline src/gatsby-theme-docz/components/NavLink/styles.js]: A brief comment would help clarity with what is going on here.

File: src/gatsby-theme-docz/wrapper.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+// The doc prop contains some metadata about the page being rendered that you can use.
+const Wrapper = ({ children }) => (
+  <>
+    <Helmet>
+      <meta charSet="utf-8" />
+      <link
+        rel="icon"
+        type="image/png"
+        href="//public/assets/favicon.png"
```
Comment (JeffMarvin) [inline src/gatsby-theme-docz/wrapper.js]: Is this the correct href?

File: src/gatsby-theme-docz/wrapper.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Helmet } from "react-helmet-async";
+
+// Note: We are using react-helmet-async here instead of react-helmet
```
Comment (JeffMarvin) [inline src/gatsby-theme-docz/wrapper.js]: The comment would help clarity if it included why this is being used.

File: src/gatsby-theme-docz/wrapper.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+// https://github.com/nfl/react-helmet/issues/426
+// https://www.docz.site/docs/add-favicon-and-metadata
+
+// eslint-disable-next-line import/no-internal-modules,no-restricted-imports
+import "../../../packages/design/src/foundation.css";
+
+// The doc prop contains some metadata about the page being rendered that you can use.
```
Comment (JeffMarvin) [inline src/gatsby-theme-docz/wrapper.js]: Is this comment still relevant to the code below?

File: src/null-markdown-loader.js [hunk 7]
```diff
@@ -0,0 +1,6 @@
+/* eslint-env node */
+
```
Comment (JeffMarvin) [inline src/null-markdown-loader.js]: A comment here explaining why this file exists would help future maintenance.

File: docs/proposals/InputDate.md [hunk 8]
```diff
@@ -6,7 +6,7 @@ The `InputDate` component allows a user to select a date.
 
 ## Wireframe
 
-![](https://d.pr/i/Eiq4KZ.png)
+![mockup](https://d.pr/i/Eiq4KZ.png)
```
[the remark this answers is not in the document]
Comment (seanhealy) [reply docs/proposals/InputDate.md]: Not currently.

File: doczrc.js [hunk 9]
```diff
@@ -1,138 +1,54 @@
...
-const modifyBundlerConfig = config => {
-  config.resolve.alias = Object.assign({}, config.resolve.alias, {
-    "@jobber/components": path.resolve(__dirname, "packages/components/src"),
-    "@jobber/docx": path.resolve(__dirname, "packages/docx/src"),
-  });
+const projectPlugin = () =>
+  createPlugin({
+    onCreateWebpackConfig: ({ stage, rules, actions, loaders, getConfig }) => {
+      // Setup CSS to allow for modules.
+      // 😢 https://github.com/gatsbyjs/gatsby/issues/16129
```
[the remark this answers is not in the document]
Comment (seanhealy) [reply doczrc.js]: The comment above was meant to be that… It's clearly not working. 😜

File: doczrc.js [hunk 10]
```diff
@@ -1,138 +1,54 @@
...
 
-const fonts = {
-  // Used for headings larger than 20px.
-  display: '"Poppins", sans-serif',
-  // Used for code and sometimes numbers in tables.
-  mono: '"Anonymous Pro", monospace',
-  // Used for text and UI (which includes almost anything).
-  ui: '"Source Sans Pro", sans-serif',
-};
+      // time-input-polyfill is incompatible with server side rendering
```
[the remark this answers is not in the document]
Comment (seanhealy) [reply doczrc.js]: Oh, this comment is out of date and should just be removed. 🤫

File: src/gatsby-theme-docz/components/Header/index.js (hunk 2 above)
Comment (seanhealy) [reply src/gatsby-theme-docz/components/Header/index.js]: This will be changing with:
https://github.com/GetJobber/atlantis/pull/217

But, perhaps some sort of documentation around component shadowing would still be good?

I don't know if this is good enough:
https://www.docz.site/docs/component-shadowing

In retrospect it makes sense but, perhaps also
https://www.gatsbyjs.org/blog/2019-04-29-component-shadowing/

To save some learning for the next person?

File: src/gatsby-theme-docz/wrapper.js (hunk 5 above)
Comment (seanhealy) [reply src/gatsby-theme-docz/wrapper.js]: The links below cover the issue. Would it be better to call them out somehow?

File: .eslintrc.js [hunk 11]
```diff
@@ -2,12 +2,31 @@ require("@jobber/eslint-config/patch-eslint-plugin-resolution.js");
...
+    },
+  },
   rules: {
     /*
       Need to figure out a good way to enforce intra vs inter module import
       rules. For now, warn on these.
      */
     "import/no-relative-parent-imports": "warn",
     "no-restricted-imports": "warn",
+    "import/no-internal-modules": [
```
[the remark this answers is not in the document]
Comment (seanhealy) [reply .eslintrc.js]: The default is error, this is relaxing it for these states.

File: doczrc.js [hunk 12]
```diff
@@ -148,6 +64,8 @@ function privateComponentReadmies() {
 // eslint-disable-next-line import/no-default-export
 export default {
   title: "🔱 Atlantis",
+  description:
+    "Atlantis is a design system for Jobber. The primary objective for Atlantis is to provide a system of reusable components to help developers to quickly build beautiful and consistent interfaces for our users.",
```
[the remark this answers is not in the document]
Comment (eddysims) [reply doczrc.js]: Confirmed that back-ticks give the line break. I think this is the way things have to be.

File: src/gatsby-theme-docz/components/Header/index.js (hunk 2 above)
Comment (seanhealy) [reply src/gatsby-theme-docz/components/Header/index.js]: Perhaps a readme in the src directory of the theme?

File: packages/generators/index.js [hunk 13]
```diff
@@ -32,6 +32,7 @@ module.exports = function(plop, config) {
       },
     ],
     actions: [
+      answers => Object.assign(answers, { mdx: "mdx" }),
```
[the remark this answers is not in the document]
Comment (seanhealy) [reply packages/generators/index.js]: Sean 👀

File: src/gatsby-theme-docz/wrapper.js (hunk 5 above)
Comment (seanhealy) [reply src/gatsby-theme-docz/wrapper.js]: Sean 👀

File: src/null-markdown-loader.js (hunk 7 above)
Comment (seanhealy) [reply src/null-markdown-loader.js]: Sean 👀

File: src/null-markdown-loader.js [hunk 14]
```diff
@@ -1,5 +1,11 @@
 /* eslint-env node */
 
+/**
+ * Null Loader works great for javascript files but, as it leaves a js
```
Comment (JeffMarvin) [inline src/null-markdown-loader.js]: the `but` goes nowhere... ?

File: src/gatsby-theme-docz/wrapper.js [hunk 15]
```diff
@@ -2,9 +2,13 @@ import React from "react";
 import PropTypes from "prop-types";
 import { Helmet } from "react-helmet-async";
 
-// Note: We are using react-helmet-async here instead of react-helmet
-// https://github.com/nfl/react-helmet/issues/426
-// https://www.docz.site/docs/add-favicon-and-metadata
+/**
+ * Note: We're referencing `react-helmet-async`, and not `react-helmet`. This
+ * is because of this (https://github.com/nfl/react-helmet/issues/426) issue.
+ * `react-helmet-async` is an API-compatible fork.
```
Comment (JeffMarvin) [inline src/gatsby-theme-docz/wrapper.js]: I guess I was hoping that we could put a short description of what that issue is (in our context), instead of making them follow the link.

Comment (seanhealy) [reply src/gatsby-theme-docz/wrapper.js]: The issue is from Docz context. So, I moved their explanation up to this level. This isn't something we've added in ourselves if that helps?

Comment (seanhealy) [reply src/gatsby-theme-docz/wrapper.js]: Changed it a bit perhaps that is better?

Comment (JeffMarvin) [reply src/gatsby-theme-docz/wrapper.js]: Looks good, thanks!
